dcp_prototype/backend/wrangling/migrations/scripts/copy_data_files.py

```diff
@@ -0,0 +1,154 @@
+import boto3
+import concurrent.futures
+import json
+import os
+import typing
+
+from argparse import ArgumentParser
+
+from boto3.s3.transfer import TransferConfig
+
+
+class FileMetadata:
+    CRC32C = "crc32c"
+    S3_ETAG = "s3_etag"
+    SHA1 = "sha1"
+    SHA256 = "sha256"
+
+
+def compose_blob_key(file_info: typing.Dict[str, str]) -> str:
+    """
+    Create the key for a blob, given the file metadata.
+    :param file_info: This can either be an object that contains the four keys (SHA256, SHA1, S3_ETAG, and CRC32C) in
+                      the key_class.
+    """
+    return "blobs/" + ".".join((
+        file_info[FileMetadata.SHA256],
+        file_info[FileMetadata.SHA1],
+        file_info[FileMetadata.S3_ETAG],
+        file_info[FileMetadata.CRC32C]
+    ))
+
+
+def list_files_in_bucket(project: str, continuation_token: str = None, keys: typing.List[str] = []) -> list:
+    """
+    Lists all files in a bucket (hard-coded to dunitz-prod-copy) with the prefix {project}/blobs
+    :param project: Name of project files belong to
+    :param continuation_token: s3 will return a max of 1000 files, pass in the continuation token to get the next 1000 for a project
+    :param keys: list of all of the file blob names
+    :return: list of all of the file blob names
+    """
+    s3 = boto3.client("s3")
+    if continuation_token:
+        response = s3.list_objects_v2(
+            Bucket='dunitz-prod-copy',
+            Prefix=f"{project}/blobs",
+            ContinuationToken=continuation_token
+        )
+    else:
+        response = s3.list_objects_v2(
+            Bucket='dunitz-prod-copy',
+            Prefix=f"{project}/blobs"
+        )
+    if response['KeyCount'] > 0:
+        for x in response['Contents']:
+            keys.append(x['Key'])
+    if response['IsTruncated'] is True:
+        list_files_in_bucket(project, response['NextContinuationToken'], keys)
+    return keys
+
+
+def copy_between_s3_buckets(key: str, project: str, max_concurrency: int = 20):
+    """
+    Copy file between s3 buckets
+    :param key: File name
+    :param project: Project name
+    :param max_concurrency: Max concurrent threads for s3 bucket, boto3 default is 10
+    """
+    config = TransferConfig(max_concurrency=max_concurrency)
+
+    s3 = boto3.resource('s3')
+    copy_source = {
+        'Bucket': 'org-hca-dss-prod',
```

Comment:
Can we move the constants to the top of the file as global variables so that it's more clear what constants' values are that it being used it code. Alternatively you could pass in these arguments (in ArgumentParser) as part of calling the script with defaults set.

Reply:
👍